tools/mcsema_disass/ida/get_cfg.py

```diff
@@ -1040,11 +1040,21 @@ def recover_regions(M, exported_vars, global_vars=[]):
     seg_name = idc.SegName(seg_ea)
     seg_names[seg_ea] = seg_name
 
-    if not is_external_segment_by_flags(seg_ea) or \
-       segment_contains_external_function_pointers(seg_ea):
+    if (not is_external_segment_by_flags(seg_ea) or \
+       segment_contains_external_function_pointers(seg_ea)) and \
+       not (is_constructor_segment(seg_ea) or is_destructor_segment(seg_ea)):
       seg_parts[seg_ea].add(seg_ea)
       seg_parts[seg_ea].add(idc.SegEnd(seg_ea))
 
+    if is_constructor_segment(seg_ea):
+      seg_parts[seg_ea].add(seg_ea)
+      end_ea =  idc.SegEnd(seg_ea)
+      if is_destructor_segment(end_ea):
+        seg_parts[seg_ea].add(idc.SegEnd(end_ea))
+      else:
+        DEBUG("WARNING: Global constructor and destructor sections are not adjacent!")
```

Comment:
I think we should be warning that they *are* adjacent, and that this special heuristic is concatenating them.

Reply:
I had the warning since the `.fini_array` was not getting handled. Updated it to handle the `.fini_array` segment if they are not adjacent and log the warning if they are adjacent.